src/test/ui/inner-static-type-parameter.rs

```diff
@@ -4,7 +4,7 @@ enum Bar<T> { What } //~ ERROR parameter `T` is never used
 
 fn foo<T>() {
     static a: Bar<T> = Bar::What;
-//~^ ERROR can't use generic parameters from outer function
+    //~^ ERROR can't use generic parameters in `static` associated item
```

Comment:
This `static` is not an associated item (and cannot be as that would entail generic statics!). It is a free item belonging to `fn foo` as a "module".